### PR TITLE
Feature/issue32 success hub recipe

### PR DESCRIPTION
**Pull request by aaronwinters**
# Critical Changes

# Changes
Completed recipe with pending objects now added

# Issues Closed
Issue 32

### CODE REVIEW

**Comment by acrosman** [comment]
Aaron thanks for working on this during the sprint. It's great (I think).  I'm definitely too tired to properly review since it requires an SSH org which I don't have handy.  If no one else gets to it soon I'll try to do so when I can.

File: snowfakery_samples/EDA/eda_student_success_hub.yml [hunk 1]
```diff
@@ -41,23 +51,6 @@
     MailingPostalCode:
       fake: postalcode
     MailingCountry: "United States"
```
Comment (eehjunggnujhee) [inline snowfakery_samples/EDA/eda_student_success_hub.yml:53]: "" is not needed.
MailingCountry: United States

File: snowfakery_samples/EDA/eda_student_success_hub.yml [hunk 2]
```diff
@@ -14,17 +17,24 @@
...
+#   Success Plan Task Template
+#   Success Plan
+#   Success Plan Task
+#   Task
+#   Alert
+#   Prediction
 
 # Run this to generate data in a scratch org:
-# cci task run generate_and_load_from_yaml -o generator_yaml snowfakery_samples/EDA/eda_student_success_hub.yml --org dev
+# cci task run snowfakery --recipe snowfakery_samples/EDA/eda_student_success_hub.yml --org dev
```
Comment (eehjunggnujhee) [inline snowfakery_samples/EDA/eda_student_success_hub.yml:29]: @aaronwinters  I added a scratch org definition for eda here. Please review the config setting for eda org that will support these recipes. https://github.com/SFDO-Community-Sprints/Snowfakery-Recipe-Templates/pull/141

We can perhaps update the target org to --org eda

File: snowfakery_samples/EDA/eda_student_success_hub.yml [hunk 3]
```diff
@@ -14,17 +17,24 @@
...
 
 # Run this to generate records in an output JSON file (NOTE: update the org_name to the CCI org you are using)
 # snowfakery ./snowfakery_samples/EDA/eda_student_success_hub.yml --output-format=json --output-file=foo.json --plugin-option org_name dev
```
Comment (eehjunggnujhee) [inline snowfakery_samples/EDA/eda_student_success_hub.yml:32]: We changed the garbage folder directory to snowfakery_samples/temp/output.json

snowfakery ./snowfakery_samples/EDA/eda_student_success_hub.yml --output-format=json --output-file=snowfakery_samples/temp/output.json --plugin-option org_name <orgAlias>

**Review by eehjunggnujhee — COMMENTED** [review]
@aaronwinters  I encountered a field level permission issue to Task.Type and access to the account EDA record types. After proper adjustment to the permission, I was able to run the recipe as described.

Screen Shot 2022-05-15 at 6 39 33 PM
Screen Shot 2022-05-15 at 6 28 06 PM

**Comment by awinters-wgu** [comment]
> @aaronwinters I encountered a field level permission issue to Task.Type and access to the account EDA record types. After proper adjustment to the permission, I was able to run the recipe as described.
> 
> Screen Shot 2022-05-15 at 6 39 33 PM
> 
> Screen Shot 2022-05-15 at 6 28 06 PM
> 
> Screen Shot 2022-05-15 at 6 46 45 PM

@eehjunggnujhee  I added a comment to the Task.Type line indicating this issue. By removing that field by default, the data generation scripts without intervention in the target org.

File: snowfakery_samples/EDA/eda_student_success_hub.yml (hunk 3 above)
Comment (awinters-wgu) [reply snowfakery_samples/EDA/eda_student_success_hub.yml:32]: @eehjunggnujhee - I updated the output file location in the recipe header. Thanks!

File: snowfakery_samples/EDA/eda_student_success_hub.yml (hunk 2 above)
Comment (awinters-wgu) [reply snowfakery_samples/EDA/eda_student_success_hub.yml:29]: I tested the commands and the following work with this recipe: 
cci flow run eda:trial_org --org dev
cci flow run sal:install_base --org dev
cci task run snowfakery --recipe snowfakery_samples/EDA/eda_student_success_hub.yml --org dev

**Comment by awinters-wgu** [comment]
@eehjunggnujhee - I added two new commits addressing the issues you raised. Thanks for your review and please let me know if you see anything else!

**Comment by aaronwinters** [comment]
@eehjunggnujhee - I added two new commits addressing the issues you raised. Thanks for your review and please let me know if you see anything else!

File: snowfakery_samples/EDA/eda_student_success_hub.yml (hunk 2 above)
Comment (aaronwinters) [reply snowfakery_samples/EDA/eda_student_success_hub.yml:29]: @eehjunggnujhee I added a comment to the Task.Type line indicating this issue. By removing that field by default, the data generation scripts without intervention in the target org.

Also, I tested the commands and the following work with this recipe:
cci flow run eda:trial_org --org dev
cci flow run sal:install_base --org dev
cci task run snowfakery --recipe snowfakery_samples/EDA/eda_student_success_hub.yml --org dev

File: snowfakery_samples/EDA/eda_student_success_hub.yml (hunk 3 above)
Comment (aaronwinters) [reply snowfakery_samples/EDA/eda_student_success_hub.yml]: @eehjunggnujhee - I updated the output file location in the recipe header. Thanks!

**Review by acrosman — APPROVED** [review]
Sorry for the _long_ delay on this.  And thank you @aaronwinters for the contributions.

**Pull request merged**